### PR TITLE
adjusts on overall features of inventory module

### PR DESCRIPTION
**Pull request by Guimino**
adjusts on overall features of inventory module

### CODE REVIEW

**Comment by Guimino** [comment]
[change "Guia de Remessa" "Requisicao Interna" and add more fields in views of Stock Adjustments](https://trello.com/c/NOA6NTnP/127-change-guia-de-remessa-requisicao-interna-and-add-more-fields-in-views-of-stock-adjustments)

File: api/src/main/java/org/openmrs/module/inventorypoc/batch/service/BatchService.java [hunk 1]
```diff
@@ -46,6 +46,8 @@
 	
 	public List<Batch> findBatchesByDrugAndLocationAndNotExpiredDate(Drug drug, Location location, Date date);
 	
+	Double findCurrentStockBalance(Drug drug, Location location, Date date);
+
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/inventorypoc/batch/service/BatchService.java]: remove empty lines

File: api/src/main/java/org/openmrs/module/inventorypoc/batch/service/BatchServiceImpl.java [hunk 2]
```diff
@@ -70,6 +70,8 @@ public Batch createBatch(final Batch batch, final BatchOperationType batchOperat
 	public void createWasteDrug(final DrugOrder drugOrder, final Location location, final Double quantity,
 	        final Date date) throws Exception {
 		
+		this.batchValidator.validateWaste(drugOrder, location, quantity, date);
+
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/inventorypoc/batch/service/BatchServiceImpl.java]: same here

File: api/src/main/java/org/openmrs/module/inventorypoc/batch/validation/BatchValidator.java [hunk 3]
```diff
@@ -22,4 +27,20 @@ public void validateAdjustments(final Batch batch, final Double newValue) {
 			    Context.getLocale()));
 		}
 	}
+	
+	public void validateWaste(final DrugOrder drugOrder, final Location location, final Double quantity,
+	        final Date date) {
+		
+		final Double currentStockBalance = Context.getService(BatchService.class)
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/inventorypoc/batch/validation/BatchValidator.java:34]: For future reference to the services classes, be initailizing them in the constructors which can be easly be proted to test classes. For example
private BatchService batchService;
public BatchValidatior(BatchService batchService){
this.batchService = batchService
}
then just use batchService as an instance **That is for later use**

File: api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteService.java [hunk 4]
```diff
@@ -51,4 +54,6 @@
 	
 	public DeliverNoteItem findDeliverNoteItemByDrugAndLotNumber(Drug drug, String lotNumber);
 	
+	public List<DeliverNoteItem> findDeliverNoteItemsByBatchAndDrugPackage(final Batch batch, final DrugPackage drugPackage);
+
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteService.java]: empty line

File: api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteServiceImpl.java [hunk 5]
```diff
@@ -182,4 +183,10 @@ public DeliverNoteItem findDeliverNoteItemByLotAndTokenNumber(final String lotNu
 	public DeliverNoteItem findDeliverNoteItemByDrugAndLotNumber(final Drug drug, final String lotNumber) {
 		return this.deliverNoteItemDAO.findByDrugAndLotNumber(drug, lotNumber, false);
 	}
+
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteServiceImpl.java:186]: empty

**Review by ningosi — COMMENTED** [review]
Please clean empty lines,  the code build successfully locally

File: api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteServiceImpl.java (hunk 5 above)
Comment (Guimino) [reply api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteServiceImpl.java:186]: @ningosi , removed empty lines.  What is the IDE are you using for java projects? Actually I'm using eclipse and there I have configured some  "Save Actions"(code style)  auto-format the code( to remove empty lines, remove unused imports, to ident the code, etc)

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/inventorypoc/delivernote/service/DeliverNoteServiceImpl.java:186]: Am using IntelliJ, now that we have a licence for it, can we all use the same IDE? or you like eclipse

**Review by ningosi — APPROVED** [review]

**Pull request merged**